tests/queries/0_stateless/02884_string_distance_function.reference

```diff
@@ -2,6 +2,8 @@ const arguments byteHammingDistance
 0
 const arguments editDistance
 6
+const arguments jaccardIndex
+0.4
 byteHammingDistance
```

Comment:
@alexey-milovidov Why did we change `byteEditDistance` -> `editDistance`, but leave `byteHammingDistance` as is? Is it intended?

Reply:
Let's put into the context that the distance functions make sense for:
- bits;
- bytes;
- UTF-8 code points;
- arrays;
- tuples.

Then, make a table with all distance functions and all types of data - where it is well-defined and where it makes sense from any practical standpoint.

Hamming distance has ambiguity between bits and bytes versions. It is often applied for bits. At the same time, the version of UTF-8 codepoints for Hamming distance can be defined but isn't practical, because it applies for vectors of the same dimension, while the UTF-8 strings are naturally very flexible in size.

Levenstein distance makes sense for bytes and UTF-8 code points, but you hardly need it for bits (it will be nonsensical).

For better context, check the names of all the existing functions working on strings...